From stern@rowland.harvard.edu  Wed Feb 20 12:55:38 2008
From: Robert Spitzenpfeil <robert@spitzenpfeil.org>
Date: Wed, 20 Feb 2008 12:11:22 -0500 (EST)
Subject: USB: usb-storage: unusual_devs entry for Oracom MP3 player
To: Greg KH <greg@kroah.com>
Cc: Robert Spitzenpfeil <robert@spitzenpfeil.org>,  Phil Dibowitz <phil@ipom.com>,  USB Storage list <usb-storage@lists.one-eyed-alien.net>
Message-ID: <Pine.LNX.4.44L0.0802201210050.5630-100000@iolanthe.rowland.org>


From: Robert Spitzenpfeil <robert@spitzenpfeil.org>

This patch (as1034) was written by Leonid Petrov, reported by Robert
Spitzenpfeil, and updated by me.  It adds an unusual_devs entry with
the IGNORE_RESIDUE flag for the Oracom MP3 player.  Together with the
change to the Get-Max-LUN routine in as1032, it makes the player usable.

Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
Signed-off-by: Phil Dibowitz <phil@ipom.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/storage/unusual_devs.h |   11 +++++++++++
 1 file changed, 11 insertions(+)

--- a/drivers/usb/storage/unusual_devs.h
+++ b/drivers/usb/storage/unusual_devs.h
@@ -1424,6 +1424,17 @@ UNUSUAL_DEV(  0x0ed1, 0x7636, 0x0103, 0x
 		US_SC_DEVICE, US_PR_DEVICE, NULL,
 		US_FL_IGNORE_RESIDUE | US_FL_GO_SLOW | US_FL_MAX_SECTORS_64),
 
+/* Patch by Leonid Petrov mail at lpetrov.net
+ * Reported by Robert Spitzenpfeil <robert@spitzenpfeil.org>
+ * http://www.qbik.ch/usb/devices/showdev.php?id=1705
+ * Updated to 103 device by MJ Ray mjr at phonecoop.coop
+ */
+UNUSUAL_DEV(  0x0f19, 0x0103, 0x0100, 0x0100,
+		"Oracom Co., Ltd",
+		"ORC-200M",
+		US_SC_DEVICE, US_PR_DEVICE, NULL,
+		US_FL_IGNORE_RESIDUE ),
+
 /* David Kuehling <dvdkhlng@gmx.de>:
  * for MP3-Player AVOX WSX-300ER (bought in Japan).  Reports lots of SCSI
  * errors when trying to write.
